src/Feature.tsx

```diff
@@ -14,157 +14,37 @@
  * limitations under the License.
  */
 import * as React from 'react'
-import { withOptimizely, WithOptimizelyProps } from './withOptimizely'
-import { VariableValuesObject, OnReadyResult, DEFAULT_ON_READY_TIMEOUT } from './client'
-import { getLogger } from '@optimizely/js-sdk-logging'
+import { UserAttributes } from '@optimizely/optimizely-sdk';
+
 
-const logger = getLogger('<OptimizelyFeature>')
+import { VariableValuesObject } from './client'
+import { useFeature } from './hooks';
+import { withOptimizely, WithOptimizelyProps } from './withOptimizely'
 
 export interface FeatureProps extends WithOptimizelyProps {
-  // TODO add support for overrideUserId
   feature: string
   timeout?: number
   autoUpdate?: boolean
+  overrideUserId?: string
+  overrideAttributes?: UserAttributes
   children: (isEnabled: boolean, variables: VariableValuesObject) => React.ReactNode
 }
 
-export interface FeatureState {
-  canRender: boolean
-  isEnabled: boolean
-  variables: VariableValuesObject
-}
-
-class FeatureComponent extends React.Component<FeatureProps, FeatureState> {
-  private optimizelyNotificationId?: number
-  private unregisterUserListener: () => void
-  private autoUpdate: boolean = false
-
-  constructor(props: FeatureProps) {
-    super(props)
-
-    this.unregisterUserListener = () => {}
-
-    const { autoUpdate, isServerSide, optimizely, feature } = props
-    this.autoUpdate = !!autoUpdate
-    if (isServerSide) {
-      if (optimizely === null) {
-        throw new Error('optimizely prop must be supplied')
-      }
-      const isEnabled = optimizely.isFeatureEnabled(feature)
-      const variables = optimizely.getFeatureVariables(feature)
-      this.state = {
-        canRender: true,
-        isEnabled,
-        variables,
-      }
-    } else {
-      this.state = {
-        canRender: false,
-        isEnabled: false,
-        variables: {},
-      }
-    }
+const FeatureComponent = (props: FeatureProps): any => {
+  const { feature, timeout, autoUpdate, children, overrideUserId, overrideAttributes } = props;
+  const [
+    isEnabled,
+    variables,
+    clientReady,
+    didTimeout
+  ] = useFeature(feature, { timeout, autoUpdate }, { overrideUserId, overrideAttributes });
```

Comment:
Should we add tests to make sure the options and overrides are passed through? Also, it seems alright to pass these even if they're not defined (looking at the tests above) but I think it could be better to construct an args array and not pass any values if they don't exist. Wdyt?

Reply:
> Should we add tests to make sure the options and overrides are passed through?

Done!

> Also, it seems alright to pass these even if they're not defined (looking at the tests above) but I think it could be better to construct an args array and not pass any values if they don't exist. Wdyt?

So 2 thoughts:

- Since these are positional args, you still need to account for the case where `userId` is not provided and `attributes` isnt
- `optimizely` here is the `ReactSDK` instance, which specifically checks for each argument to be `undefined`.

As such, I think this is the best approach.